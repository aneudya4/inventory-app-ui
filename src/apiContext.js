import React, { useContext } from 'react';

export const useApiContext = () => useContext(ApiContext);
const ApiContext = React.createContext({
  products: [],
  orders: [],
  user: null,
  setUser: () => {},
});

export default ApiContext;
