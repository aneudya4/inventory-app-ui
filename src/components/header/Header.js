import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import './header.css';
import ApiContext from '../../apiContext';
import app from '../../firebaseConfig';

const Header = (props) => {
  const { user } = useContext(ApiContext);
  const renderAuthNav = () => {
    if (!user) {
      return (
        <>
          <li>
            <Link to='/accounts/register'>Register</Link>
          </li>
          <li>
            <Link to='/accounts/login'>Log In</Link>
          </li>
        </>
      );
    }
    return (
      <>
        <li>
          <Link to='/auth/dashboard/overview'>Dashboard</Link>
        </li>
        <li>
          <Link onClick={() => app.auth().signOut()} to='/'>
            Logout
          </Link>
        </li>
      </>
    );
  };
  return (
    <header>
      <Link to='/'>
        <h1>Invy</h1>
      </Link>
      <nav>
        <ul>{renderAuthNav()}</ul>
      </nav>
    </header>
  );
};

export default Header;
