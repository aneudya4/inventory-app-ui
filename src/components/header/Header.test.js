import { shallow } from 'enzyme';
import * as ApiContext from '../../apiContext';
import Header from './Header';

describe('Test on the <Header/> Component', () => {
  it('renders Header without crashing', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
