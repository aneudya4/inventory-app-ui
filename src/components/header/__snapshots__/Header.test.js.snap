// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on the <Header/> Component renders Header without crashing 1`] = `
<header>
  <Link
    to="/"
  >
    <h1>
      Invy
    </h1>
  </Link>
  <nav>
    <ul>
      <li>
        <Link
          to="/accounts/register"
        >
          Register
        </Link>
      </li>
      <li>
        <Link
          to="/accounts/login"
        >
          Login
        </Link>
      </li>
    </ul>
  </nav>
</header>
`;
